homeassistant/components/picnic/config_flow.py

```diff
@@ -0,0 +1,112 @@
+"""Config flow for Picnic integration."""
+import logging
+
+from python_picnic_api import PicnicAPI
+from python_picnic_api.session import PicnicAuthError
+import requests
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import (  # pylint: disable=unused-import
+    CONF_COUNTRY_CODE,
+    COUNTRY_CODES,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required(CONF_COUNTRY_CODE, default=COUNTRY_CODES[0]): vol.In(
+            COUNTRY_CODES
+        ),
+    }
+)
+
+
+class PicnicHub:
+    """Hub class to test user authentication."""
+
+    @staticmethod
+    def authenticate(username, password, country_code) -> None:
+        """Test if we can authenticate with the Picnic API."""
+        picnic = PicnicAPI(username, password, country_code)
```

Comment:
Was looking at the code of this integration.

The token you get here <https://github.com/MikeBrink/python-picnic-api/blob/master/python_picnic_api/session.py#L40>, how long is that token valid? I am assuming the mobile app is not requesting the auth the whole time. Would you be able to get away with just taking this auth token and not storing the username/password the whole time?

Reply:
Here is a way it can be structured so that if the token is refreshed, it will be able to be updated in HA too https://developers.home-assistant.io/docs/api_lib_auth

Reply:
I've tested with an auth token I'd saved a while back and I could still use it to authenticate, so it's possible indeed to just save the token. I'll update this as soon as MikeBrink/python-picnic-api#11 is merged and released.

Reply:
I've updated the integration so it saves the authentication token instead of the username password. It also updates the token in the config entry when an updated one is sent by Picnic.